qa/tasks/cephfs/test_misc.py

```diff
@@ -198,6 +198,10 @@ def test_dump_inode_hexademical(self):
         info = self.fs.mds_asok(['dump', 'inode', hex(ino)])
         assert info['path'] == "/foo"
 
+    def test_dispatch_queue_throttle(self):
+        with self.assert_cluster_log("Throttler Limit has been hit. Some message processing may be significantly delayed.",
```

Comment:
I assume you'll need to set certain `ms_` throttler options to replicate the behavior. Then hit the MDS with a lot of client traffic by e.g. creating some directory tree.

Reply:
I believe we could use either default throttle options or custom values. Looking more on this today, I think we should implement the test as part of src/test/msgr/test_msgr.cc? I was looking at [this test](https://github.com/ceph/ceph/blob/master/src/test/msgr/test_msgr.cc#L2051) in particular and the workload it creates for testing. Suggestions?

Reply:
You could put the test there. I've not looked at these tests before so I have no specific suggestions. You'll need to think about how you want to detect the throttling as part of the test to confirm your code functions.